plasmapy/plasma/fields.py

```diff
@@ -0,0 +1,326 @@
+        self._bmax = bmax
+
+        self.E = np.zeros(self.grid.shape) * u.V / u.m
+        self.B = np.zeros(self.grid.shape) * u.T
+
+        # Calculate a characteristic length scale from the grid to scale
+        # field features with
+        self.L = np.abs(np.max(self.grid.grid) - np.min(self.grid.grid))
+
+        # Calculate radius arrays in spherical and cylindrical coordinates
+        # for use in generating different field structures
+        self.radius = np.sqrt(
+            self.grid.x_arr ** 2 + self.grid.y_arr ** 2 + self.grid.z_arr ** 2
+        )
+        self.pradius = np.sqrt(self.grid.x_arr ** 2 + self.grid.y_arr ** 2)
+
+        # Check that the model selected can be used (eg. no gradients on a
+        # non-uniform grid)
+        self._validate()
+
+        # Generate the fields.
+        self._gen_fields()
+
+        # Normalize the fields to the emax and bmax values set
+        self._norm()
+
+    @abstractmethod
+    def _validate(self):
+        """
+        Raise errors when the specified parameteters are not compatible with
+        the field subclass that is being instantiated.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def _gen_fields(self):
+        """
+        Fill the empty E and B arrays with the example fields.
+        """
+        raise NotImplementedError
+
+    def _norm(self):
+        """
+        Normalize the fields to the emax and bmax values provided.
+        """
+        max_E = np.max(self.E)
+        max_B = np.max(self.B)
+
+        if max_E != 0:
+            self.E = self._emax * (self.E / max_E).to(u.dimensionless_unscaled)
+
+        if max_B != 0:
+            self.B = self._bmax * (self.B / max_B).to(u.dimensionless_unscaled)
+
+    @property
+    def bmax(self):
+        """
+        Get bmax
+        """
+        return self._bmax
+
+    @bmax.setter
+    def bmax(self, val):
+        """
+        Set bmax and re-normalize fields to the new value.
+        """
+        self._bmax = val
+        # Normalize the fields to the new value
+        self._norm()
+
+    @property
+    def emax(self):
+        """
+        Get emax
+        """
+        return self._emax
+
+    @emax.setter
+    def emax(self, val):
+        """
+        Set emax and re-normalize fields to the new value.
+        """
+        self._emax = val
+        # Normalize the fields to the new values
+        self._norm()
+
+
+class NoFields(AbstractField):
+    r"""
+    Empty E and B arrays.
+    """
+
+    def _gen_fields(self):
+        pass
+
+    def _validate(self):
+        pass
+
+
+class ElectrostaticGaussianSphere(AbstractField):
+    r"""
+    A radial, spherically symmetric electric field produced by a spherical
+    blob of potential with a Gaussian radial distribution:
+
+    .. math::
+        \phi = e^{-(r/a)^2}
+    .. math::
+        E = -\nabla \phi \text{, for }  (r < L/2)
+    .. math::
+        E = 0 \text{, for } (r < L/2)
+    .. math::
+        B = 0
```

Comment:
I'm no .rst wizard, but - couldn't this go into a single `.. math::` block?

Reply:
I had it like that initially and had some formatting problems. It makes sense it should work that way, though. If you're not quite an `.rst` wizard I'm more like an `.rst` gnome or something.